lang/csharp/src/apache/main/Reflect/ClassCache.cs

```diff
@@ -48,13 +52,30 @@ private void AddClassNameMapItem(RecordSchema schema, Type dotnetClass)
             _nameClassMap.TryAdd(schema.Fullname, new DotnetClass(dotnetClass, schema, this));
         }
 
+        /// <summary>
+        /// Clears all the default converters for testing purposes
+        /// </summary>
+        internal static List<IAvroFieldConverter> ClearDefaultConverters()
+        {
+            var result = new List<IAvroFieldConverter>();
+            while (_defaultConverters.TryTake(out IAvroFieldConverter item))
+            {
+                result.Add(item);
+            }
+
+            _typeDefaultConverters.Clear();
+
+            return result;
+        }
+
         /// <summary>
         /// Add a default field converter
         /// </summary>
         /// <param name="converter"></param>
         public static void AddDefaultConverter(IAvroFieldConverter converter)
         {
             _defaultConverters.Add(converter);
+            _typeDefaultConverters.Clear();
```

Comment:
Why?

Reply:
I'll add a comment, but when a new converter is added, it should flush the cached default converters and check if the new converter is valid for some previous case where no converter was found.  This feels like an edge case - but it might be something that happens during some client code's unit tests.  IRL, I would expect that all the default converters are added before any serialization occurs.  But, in a testing world?  I'm betting it won't be the case.